lib/GIL/BasicBlock.cpp

```diff
@@ -53,28 +52,16 @@ void BasicBlock::removeInstruction(InstBase *inst)
     _instructions.remove(inst);
 }
 
-InstBase *BasicBlock::getTerminator()
-{
-    if (_instructions.empty() /*|| !_instructions.back().isTerminator()*/)
-        return nullptr;
-    return &_instructions.back();
-}
-
 TerminatorInst *BasicBlock::getTerminatorInst()
 {
-    InstBase *terminator = getTerminator();
-    if (terminator) {
-        return static_cast<TerminatorInst *>(terminator);
-    }
-    return nullptr;
+    if (_instructions.empty())
+        return nullptr;
+    return llvm::dyn_cast<TerminatorInst>(&_instructions.back());
 }
 
 void BasicBlock::setTerminator(InstBase *terminator)
 {
-    InstBase *lastInst = getTerminator();
-    bool lastInstIsTerminator = lastInst != nullptr;
-
-    if (lastInstIsTerminator) {
+    if (TerminatorInst *lastInst = getTerminatorInst()) {
```

Comment:
I don't think it's pretty

Reply:
why write 4 lines when it can be 1 line? that's just the syntax to unwrap an optional / check for null in C++